test/user_test.rb

```diff
@@ -0,0 +1,51 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+  def setup
+    @sal = User.new("Sal")
+    @ali = User.new("Ali")
+    @joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    @joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+  end
+
+  def test_it_exists
+
+    assert_instance_of User, @sal
+  end
+
+  def test_it_has_attributes
+
+    assert_equal "Sal", @sal.name
+    assert_equal [], @sal.jokes
+  end
+
+  def test_user_can_learn_jokes
+    @sal.learn(@joke_1)
+    @sal.learn(@joke_2)
+
+    assert_equal [@joke_1, @joke_2], @sal.jokes
+  end
+
+  def test_user_can_tell_jokes
+    @sal.learn(@joke_1)
+    @sal.learn(@joke_2)
+    @sal.tell(@ali, @joke_1)
+    @sal.tell(@ali, @joke_2)
+
+    assert_equal [@joke_1, @joke_2], @ali.jokes
```

Comment:
great job making the association that a user must learn a joke before they can tell the joke